lib/libyui_client/validate.rb

```diff
@@ -5,7 +5,8 @@ module Validate
     module_function
 
     def filter(filter)
-      filters = %i[id label type]
+      # 'id', 'label', and 'type' are server side filters while 'regexp' is a client side filter
+      filters = %i[id label type regexp]
```

Comment:
Hmm, not sure if that's exactly what we need. I would say that reg exp (or something with wildcards) is applicable for search by label and by id. And I would not create a separate field for it.
What we could do is check the class of the parameter, e.g.
`label.is_a?(Regexp)` and then treat it as a regexp doing filtering on the client side. We also could move all filtering for the reading values to the client side, so that we request whole json and find widgets we need. This will create some network overhead, but we can implement compressing and caching once it becomes an issue.

Reply:
After checking Regexp class, we can actually call `match` method on both strings and regexps, so we don't need special behavior for those 2 cases. The only thing we should be sure about is that match is 100% and not just a part of the string, as e.g. `"hello".match("hell")` will partly match as well as `"hello".match(/./)`.

Reply:
discussed offline that we address this in near future.